controllers/components/kserve/kserve_support.go

```diff
@@ -0,0 +1,302 @@
+package kserve
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"path"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/manifest"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/serverless"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/servicemesh"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+)
+
+var serviceAccounts = map[cluster.Platform][]string{
+	cluster.Unknown:          {odhModelControllerComponentName},
+	cluster.OpenDataHub:      {odhModelControllerComponentName},
+	cluster.ManagedRhods:     {odhModelControllerComponentName},
+	cluster.SelfManagedRhods: {odhModelControllerComponentName},
+}
+
+// clusterRoleAggregationRulePredicate is a watch predicate that can be used with
+// ClusterRoles to ignore the rules field on update if aggregationRule is set.
+var clusterRoleAggregationRulePredicate = predicate.Funcs{
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		oldClusterRole, ok := e.ObjectOld.DeepCopyObject().(*rbacv1.ClusterRole)
+		if !ok {
+			return true
+		}
+		newClusterRole, ok := e.ObjectNew.DeepCopyObject().(*rbacv1.ClusterRole)
+		if !ok {
+			return true
+		}
+
+		// if aggregationRule is set, then the rules are set by k8s
+		// based on other ClusterRoles matching a label selector, so we
+		// shouldn't try to reset that back to empty
+		if newClusterRole.AggregationRule != nil {
+			oldClusterRole.Rules = nil
+			newClusterRole.Rules = nil
```

Comment:
I'd probably move keep the 2 lines above as part of the if and then move the lines below outside.

So basically, if it is an aggregation, discard rules from comparison, but still perform a comparison on the other fields so i.e. we can revert it in case one changes the aggregation rule itself.